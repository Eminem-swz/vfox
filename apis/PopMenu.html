<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>PopMenu 气泡菜单 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.cd17bc9c.css">
    <link rel="modulepreload" href="/vfox/assets/Home.1e98aa66.js">
    <link rel="modulepreload" href="/vfox/assets/app.082b569a.js">
    <link rel="modulepreload" href="/vfox/assets/apis_PopMenu.md.62020e15.lean.js">
    
    <meta name="twitter:title" content="PopMenu 气泡菜单 | Vfox for Vue3">
  <meta property="og:title" content="PopMenu 气泡菜单 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-cbbf79d8><div class="sidebar-button" data-v-cbbf79d8><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-cbbf79d8 data-v-3902cc82><!----> Vfox for Vue3</a><div class="flex-grow" data-v-cbbf79d8></div><div class="nav" data-v-cbbf79d8><nav class="nav-links" data-v-cbbf79d8 data-v-6c4348e2><!--[--><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/" data-v-3efd0139>指南 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/design/" data-v-3efd0139>设计 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/components/Button" data-v-3efd0139>组件 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item active" href="/vfox/apis/Toast" data-v-3efd0139>API <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-dropdown-link" data-v-6c4348e2 data-v-a5ec8504><button class="button" data-v-a5ec8504><span class="button-text" data-v-a5ec8504>选择版本</span><span class="right button-arrow" data-v-a5ec8504></span></button><ul class="dialog" data-v-a5ec8504><!--[--><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>1.x</span><span class="icon" data-v-5a696df7><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-5a696df7><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item active" href="/vfox/" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>2.x</span><span class="icon" data-v-5a696df7><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-722cd14b><nav class="nav-links nav" data-v-722cd14b data-v-6c4348e2><!--[--><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/" data-v-3efd0139>指南 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/design/" data-v-3efd0139>设计 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/components/Button" data-v-3efd0139>组件 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item active" href="/vfox/apis/Toast" data-v-3efd0139>API <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-dropdown-link" data-v-6c4348e2 data-v-a5ec8504><button class="button" data-v-a5ec8504><span class="button-text" data-v-a5ec8504>选择版本</span><span class="right button-arrow" data-v-a5ec8504></span></button><ul class="dialog" data-v-a5ec8504><!--[--><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>1.x</span><span class="icon" data-v-5a696df7><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-5a696df7><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item active" href="/vfox/" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>2.x</span><span class="icon" data-v-5a696df7><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-722cd14b><!--[--><li class="sidebar-link"><p class="sidebar-link-item">通知类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Notify">Notify</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">预览类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Preview">Preview</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">模态框类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Dialog">Dialog</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">抽屉类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Calendar">Calendar</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">气泡类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/apis/PopMenu">PopMenu</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#showpopmenu-object">showPopMenu(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage">Usage</a><!----></li></ul></li></ul></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-0fc60803><div class="container" data-v-0fc60803><!--[--><!--[--><!----><!--]--><!--[--><div class="demo"><iframe style="display:none;" src="https://godxiaoji.github.io/vfox/demo/#/" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-0fc60803><div><h1 id="popmenu-气泡菜单" tabindex="-1">PopMenu 气泡菜单 <a class="header-anchor" href="#popmenu-气泡菜单" aria-hidden="true">#</a></h1><h2 id="showpopmenu-object" tabindex="-1">showPopMenu(object) <a class="header-anchor" href="#showpopmenu-object" aria-hidden="true">#</a></h2><p>显示气泡菜单。</p><h3 id="object" tabindex="-1">object <a class="header-anchor" href="#object" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>selector</td><td>string | HTMLElement</td><td></td><td>是</td><td>从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到</td></tr><tr><td>placement</td><td>string</td><td>&#39;bottom&#39;</td><td>否</td><td>展开位置，可选 &#39;bottom&#39;, &#39;top&#39;, &#39;left&#39;, &#39;right&#39;</td></tr><tr><td>options</td><td>array</td><td></td><td>否</td><td>选项列表</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，选择选项）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h4 id="options-的结构" tabindex="-1">options 的结构 <a class="header-anchor" href="#options-的结构" aria-hidden="true">#</a></h4><table><thead><tr><th>options[index] 值</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>name</td><td>string</td><td></td><td>是</td><td>选项名</td></tr><tr><td>disabled</td><td>string</td><td>false</td><td>否</td><td>是否禁用</td></tr><tr><td>icon</td><td>string</td><td></td><td>否</td><td>图标，使用 <a href="./../components/Icon.html">Icon</a> 组件</td></tr></tbody></table><div class="language-"><pre><code>[
  {
    name: &#39;选项1&#39;,
    disabled: false,
    icon: &#39;MenuOutlined&#39;
  }
]
</code></pre></div><h4 id="object-success-回调参数" tabindex="-1">object.success 回调参数 <a class="header-anchor" href="#object-success-回调参数" aria-hidden="true">#</a></h4><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示用户点击了选项，此时返回 <code>detail</code></td></tr><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示用户点击了取消</td></tr><tr><td>detail?.item</td><td>any</td><td>options[index] 的副本</td></tr><tr><td>detail?.index</td><td>number</td><td>索引</td></tr></tbody></table><h3 id="usage" tabindex="-1">Usage <a class="header-anchor" href="#usage" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">showPopMenu</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  <span class="token literal-property property">selector</span><span class="token operator">:</span> <span class="token string">&#39;#popMenuTarget&#39;</span><span class="token punctuation">,</span>
  <span class="token literal-property property">options</span><span class="token operator">:</span> <span class="token punctuation">[</span><span class="token punctuation">{</span>
    <span class="token literal-property property">icon</span><span class="token operator">:</span> <span class="token string">&#39;HeartOutlined&#39;</span><span class="token punctuation">,</span>
    <span class="token literal-property property">name</span><span class="token operator">:</span> <span class="token string">&#39;爱心&#39;</span>
  <span class="token punctuation">}</span><span class="token punctuation">,</span>
  <span class="token punctuation">{</span>
    <span class="token literal-property property">icon</span><span class="token operator">:</span> <span class="token string">&#39;StarOutlined&#39;</span><span class="token punctuation">,</span>
    <span class="token literal-property property">name</span><span class="token operator">:</span> <span class="token string">&#39;星星&#39;</span>
  <span class="token punctuation">}</span><span class="token punctuation">,</span>
  <span class="token punctuation">{</span>
    <span class="token literal-property property">icon</span><span class="token operator">:</span> <span class="token string">&#39;CircleOutlined&#39;</span><span class="token punctuation">,</span>
    <span class="token literal-property property">name</span><span class="token operator">:</span> <span class="token string">&#39;圈圈&#39;</span><span class="token punctuation">,</span>
    <span class="token literal-property property">disabled</span><span class="token operator">:</span> <span class="token boolean">true</span>
  <span class="token punctuation">}</span><span class="token punctuation">]</span><span class="token punctuation">,</span>
  <span class="token literal-property property">placement</span><span class="token operator">:</span> <span class="token string">&#39;top&#39;</span><span class="token punctuation">,</span>
  <span class="token function-variable function">success</span><span class="token operator">:</span> <span class="token parameter">res</span> <span class="token operator">=&gt;</span> <span class="token punctuation">{</span>
    console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">&#39;select&#39;</span><span class="token punctuation">,</span> res<span class="token punctuation">)</span>
  <span class="token punctuation">}</span>
<span class="token punctuation">}</span><span class="token punctuation">)</span>
</code></pre></div></div></div><footer class="page-footer" data-v-0fc60803 data-v-e4f5d8c8><div class="edit" data-v-e4f5d8c8><div class="edit-link" data-v-e4f5d8c8 data-v-25f5f3d4><!----></div></div><div class="updated" data-v-e4f5d8c8><!----></div></footer><div class="next-and-prev-link" data-v-0fc60803 data-v-162a74c0><div class="container" data-v-162a74c0><div class="prev" data-v-162a74c0><a class="link" href="/vfox/apis/PopDialog" data-v-162a74c0><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-162a74c0><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-162a74c0>PopDialog</span></a></div><div class="next" data-v-162a74c0><!----></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_actionsheet.md\":\"b1f641b5\",\"apis_calendar.md\":\"fee32384\",\"apis_cascader.md\":\"5f6b8223\",\"apis_datepicker.md\":\"21544010\",\"apis_dialog.md\":\"0c7da301\",\"apis_notify.md\":\"c568a5bb\",\"apis_picker.md\":\"8ee62906\",\"apis_popdialog.md\":\"c990ec08\",\"apis_popmenu.md\":\"62020e15\",\"apis_popover.md\":\"281635e2\",\"apis_preview.md\":\"d71c7234\",\"apis_toast.md\":\"ddc8ec4a\",\"apis_index.md\":\"b4151466\",\"components_actionsheet.md\":\"513be7ff\",\"components_activityindicator.md\":\"3a4f4015\",\"components_avatar.md\":\"b771f884\",\"components_backtop.md\":\"7d8810e9\",\"components_badge.md\":\"69dd086f\",\"components_button.md\":\"c5dcd98c\",\"components_calendar.md\":\"4b58268d\",\"components_cascader.md\":\"500f7f2a\",\"components_cell.md\":\"7107bdcc\",\"components_checkbox.md\":\"9579681c\",\"components_circleprogress.md\":\"131949e0\",\"components_collapse.md\":\"8669e3bd\",\"components_copy.md\":\"5f69b695\",\"components_countdown.md\":\"d96583c2\",\"components_countup.md\":\"21c053fb\",\"components_datepicker.md\":\"543f88b9\",\"components_dialog.md\":\"285d66d7\",\"components_divider.md\":\"40e83018\",\"components_drawer.md\":\"50be4b35\",\"components_dropdown.md\":\"585a1ba7\",\"components_empty.md\":\"bdd7b844\",\"components_fixed.md\":\"b4589a75\",\"components_flatlist.md\":\"c8776885\",\"components_form.md\":\"b6d11440\",\"components_group.md\":\"5e0b5b8d\",\"components_icon.md\":\"a9f911d5\",\"components_image.md\":\"bff386fd\",\"components_imagepreview.md\":\"6f448b31\",\"components_imageuploader.md\":\"d5704395\",\"components_indexview.md\":\"f6003376\",\"components_input.md\":\"522f1959\",\"components_layout.md\":\"cd84ae87\",\"components_loadmore.md\":\"a107dc74\",\"components_modal.md\":\"906ac40b\",\"components_navbar.md\":\"22ee6584\",\"components_noticebar.md\":\"2324f3ce\",\"components_notify.md\":\"ddb38c80\",\"components_numberkeyboard.md\":\"689fa603\",\"components_order.md\":\"34cb876a\",\"components_pagination.md\":\"a5972166\",\"components_picker.md\":\"ccea1a25\",\"components_popdialog.md\":\"0db87cb2\",\"components_popmenu.md\":\"61d3251e\",\"components_popover.md\":\"9d922f7d\",\"components_price.md\":\"bbb36d36\",\"components_progress.md\":\"071efebf\",\"components_radio.md\":\"1200ff8d\",\"components_range.md\":\"bcaa7031\",\"components_rate.md\":\"2fd4c9c2\",\"components_result.md\":\"73290448\",\"components_scrolltab.md\":\"5e7ce23d\",\"components_scrollview.md\":\"210e6a34\",\"components_searchbar.md\":\"18e06c52\",\"components_sidetab.md\":\"a6822f8e\",\"components_skeleton.md\":\"43743347\",\"components_slider.md\":\"30a68334\",\"components_stepper.md\":\"3a8b4b74\",\"components_steps.md\":\"eb74e87b\",\"components_sticky.md\":\"43f335bf\",\"components_stopwatch.md\":\"acd9c3eb\",\"components_swipecell.md\":\"f6f9cd4e\",\"components_swiper.md\":\"f72fe0c0\",\"components_switch.md\":\"2fb300d3\",\"components_tab.md\":\"a9907474\",\"components_tabbar.md\":\"c39bf75a\",\"components_tabview.md\":\"b6832fcd\",\"components_tag.md\":\"1d16063d\",\"components_timeago.md\":\"debba5ad\",\"components_timeline.md\":\"210aabb7\",\"components_toast.md\":\"aa4700f0\",\"components_index.md\":\"a3ace032\",\"design_color.md\":\"6e7847a8\",\"design_index.md\":\"e2c76f01\",\"guide_dark.md\":\"559e4331\",\"guide_import.md\":\"9a552115\",\"guide_locale.md\":\"49a6428d\",\"guide_theme.md\":\"dc2cd3aa\",\"index.md\":\"6e5d562e\"}")</script>
    <script type="module" async src="/vfox/assets/app.082b569a.js"></script>
    
  </body>
</html>