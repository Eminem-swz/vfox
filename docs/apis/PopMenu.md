# PopMenu 气泡菜单

## showPopMenu(object)

显示气泡菜单。

### object

| 属性      | 类型           | 默认值   | 必填 | 说明                                                                              |
| --------- | -------------- | -------- | ---- | --------------------------------------------------------------------------------- |
| selector  | string/element |          | 是   | 从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到 |
| placement | string         | 'bottom' | 否   | 展开位置，可选 'bottom', 'top', 'left', 'right'                                   |
| options   | array          |          | 否   | 选项列表                                                                          |
| success   | function       |          | 否   | 接口调用成功（在用户做出选择后，如取消，选择选项）的回调函数                      |
| fail      | function       |          | 否   | 接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）          |
| complete  | function       |          | 否   | 弹窗关闭或调用失败的回调函数                                                      |

#### options 的结构

| options[index] 值 | 类型   | 默认值 | 必填 | 说明                                          |
| ----------------- | ------ | ------ | ---- | --------------------------------------------- |
| name              | string |        | 是   | 选项名                                        |
| disabled          | string | false  | 否   | 是否禁用                                      |
| icon              | string |        | 否   | 图标，使用 [Icon](../components/Icon.md) 组件 |

```
[
  {
    name: '选项1',
    disabled: false,
    icon: 'MenuOutlined'
  }
]
```

#### object.success 回调参数

| 属性          | 类型    | 说明                                              |
| ------------- | ------- | ------------------------------------------------- |
| confirm?      | boolean | 为 true 时，表示用户点击了选项，此时返回 `detail` |
| cancel?       | boolean | 为 true 时，表示用户点击了取消                    |
| detail?.item  | any     | options[index] 的副本                             |
| detail?.index | number  | 索引                                              |

### Usage

具体调用方式可以参考[API 调用](../guide/import.md#api-调用)。

```JavaScript
showPopMenu({
  selector: '#popMenuTarget',
  options: [{
    icon: 'HeartOutlined',
    name: '爱心'
  },
  {
    icon: 'StarOutlined',
    name: '星星'
  },
  {
    icon: 'CircleOutlined',
    name: '圈圈',
    disabled: true
  }],
  placement: 'top',
  success: res => {
    console.log('select', res)
  }
})
```
