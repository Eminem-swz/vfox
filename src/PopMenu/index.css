.fx-pop-menu .fx-mask {
  background-color: rgba(0, 0, 0, 0.1);
}

.fx-pop-menu_list {
  list-style: none;
  padding: 0;
  margin: 0;
  background-color: #fff;
}

.fx-pop-menu_item {
  cursor: pointer;
}

.fx-pop-menu_item:not(.disabled):active {
  background-color: rgba(0, 0, 0, 0.16);
}

.fx-pop-menu_item:not(:last-child)::after {
  content: '';
}

.fx-pop-menu_item-inner {
  height: 48px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  line-height: 22px;
  color: #262626;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0 24px;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.fx-pop-menu_item-inner .fx-icon {
  --size: 18px;
  --color: #262626;
  vertical-align: middle;
  margin: 0 12px 0 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.fx-pop-menu_item-inner span {
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  overflow: hidden;
}

.fx-pop-menu_item-inner.has--icon {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0 20px;
}

.fx-pop-menu_item.disabled {
  cursor: not-allowed;
  opacity: 0.2;
}
