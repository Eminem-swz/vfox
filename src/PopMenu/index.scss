@import '../style/var.scss';

.fx-pop-menu {
  .fx-mask {
    background-color: rgba($color: $black-color, $alpha: 0.1);
  }

  &_list {
    list-style: none;
    padding: 0;
    margin: 0;
    background-color: #fff;
  }

  &_item {
    cursor: pointer;

    &:not(.disabled):active {
      @include background-active;
    }

    &:not(:last-child)::after {
      content: '';
    }

    &-inner {
      height: 48px;
      display: flex;
      align-items: center;
      font-size: 16px;
      line-height: 22px;
      color: $title-color;
      text-align: center;
      user-select: none;
      padding: 0 24px;
      box-sizing: border-box;
      justify-content: center;

      .fx-icon {
        --size: 18px;
        --color: #{$title-color};
        vertical-align: middle;
        margin: 0 12px 0 0;
        flex-shrink: 0;
      }

      span {
        display: -webkit-box;
        -webkit-line-clamp: 1;
        -webkit-box-orient: vertical;
        flex-shrink: 1;
        overflow: hidden;
      }

      &.has--icon {
        justify-content: flex-start;
        padding: 0 20px;
      }
    }

    &.disabled {
      cursor: not-allowed;
      opacity: 0.2;
    }
  }
}
