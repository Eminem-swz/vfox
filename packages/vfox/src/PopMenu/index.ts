import { withInstall } from '../helpers/with-install'
import PopMenu from '../Popover/PopMenu.vue'
import { createConfirmHook, createShowPopup } from '../popup/api'
import type { PopupSuccessConfirmArgs } from '../popup/types'
import type { DomSelector, PlacementType } from '../helpers/types'
import type { MenuOption, MenuDetail } from '../Popover/types'

const showPopMenu = createShowPopup<
  {
    selector: DomSelector
    options: MenuOption[]
    placement?: PlacementType
  },
  PopupSuccessConfirmArgs<MenuDetail>
>({
  apiName: 'showPopMenu',
  component: PopMenu,
  createHook: createConfirmHook
})

export { PopMenu, showPopMenu }
export default PopMenu
export const install = withInstall(PopMenu)
