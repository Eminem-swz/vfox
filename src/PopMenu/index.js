import PopMenu from '../Popover/PopMenu.vue'
import { showPopMenu } from '../apis/Popup'

PopMenu.install = function(app) {
  app.component(PopMenu.name, PopMenu)

  // 添加api
  app.config.globalProperties.$showPopMenu = showPopMenu
}

PopMenu.showPopMenu = showPopMenu

export default PopMenu
