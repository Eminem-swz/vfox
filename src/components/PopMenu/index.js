import PopMenu from '../Popover/Menu.vue'

PopMenu.install = function(Vue) {
  Vue.component(PopMenu.name, PopMenu)
}

export default PopMenu
