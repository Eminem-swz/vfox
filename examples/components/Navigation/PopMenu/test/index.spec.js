/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import PopMenu from '@/components/PopMenu'

const options = [
  {
    icon: 'HeartOutlined',
    name: '爱心'
  },
  {
    icon: 'StarOutlined',
    name: '星星'
  },
  {
    icon: 'CircleOutlined',
    name: '圈圈',
    disabled: true
  }
]

describe('PopMenu', () => {
  test('should render prop set options correctly', () => {
    const wrapper = mount(PopMenu, {
      propsData: {
        options
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
