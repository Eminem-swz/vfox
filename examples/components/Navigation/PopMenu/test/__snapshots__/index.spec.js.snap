// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopMenu should render prop set options correctly 1`] = `
<div class="fx-popover fx-pop-menu fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-mask"></div>
  <div class="fx-popover_inner" style="left: 0px; top: 0px; max-width: -16px; max-height: -16px;"><i class="fx-popover_arrow" style="left: 200vw; top: 200vh;"></i>
    <div class="fx-popover_content">
      <ul class="fx-pop-menu_list fx-pop-menu_list">
        <li class="fx-pop-menu_item fx-horizontal-hairline">
          <div class="fx-pop-menu_item-inner has--icon"><svg class="fx-icon">
              <use xlink:href="#icon-HeartOutlined"></use>
            </svg> <span>爱心</span></div>
        </li>
        <li class="fx-pop-menu_item fx-horizontal-hairline">
          <div class="fx-pop-menu_item-inner has--icon"><svg class="fx-icon">
              <use xlink:href="#icon-StarOutlined"></use>
            </svg> <span>星星</span></div>
        </li>
        <li class="fx-pop-menu_item fx-horizontal-hairline disabled">
          <div class="fx-pop-menu_item-inner has--icon"><svg class="fx-icon">
              <use xlink:href="#icon-CircleOutlined"></use>
            </svg> <span>圈圈</span></div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
